core/server/web/shared/middlewares/brute.js

```diff
@@ -73,5 +73,26 @@ module.exports = {
                 next('privateblog');
             }
         })(req, res, next);
+    },
+
+    /*
+     * protect content api from brute force
+     */
+    contentApiKey(req, res, next) {
+        return spamPrevention.contentApiKey().getMiddleware({
+            ignoreIP: false
+        })(req, res, function (err, ...rest) {
```

Comment:
Here we make sure not to add the `finish` listener, unless
a) the spamPrevention middleware has finished
b) it has finished without an error